### PR TITLE
Share-logs GCE update

### PR DESCRIPTION
**Pull request by markvoll**
User is now notified if the selected bucket
is not in the correct project or can not be created.

### CODE REVIEW

File: brkt_cli/gce/gce_service.py [hunk 1]
```diff
@@ -219,52 +219,32 @@ def __init__(self, project, session_id, logger):
...
         except errors.HttpError as e:
-            code = e.resp.status
-            if code == 404:
-                # bucket doesn't exist. Check syntax
-                self.validate_bucket_name(bucket)
-            elif code == 403:
-                # bucket does exist. Permisions not valid
-                raise ValidationError("Permission denied for bucket %s" % bucket)
+            message = json.loads(e.content)["error"]["message"]
+            if 'You already own this bucket' in message:
```
Comment (bburtin) [inline brkt_cli/gce/gce_service.py]: Does GCE give you an error code?  It's a bit brittle to make assumptions about the message text.

File: brkt_cli/gce/gce_service.py [hunk 2]
```diff
@@ -219,52 +219,32 @@ def __init__(self, project, session_id, logger):
...
+                return
             else:
-                # unexpected Http error
+                self.log.warn("Bucket must be in project: %s", project)
```
Comment (bburtin) [inline brkt_cli/gce/gce_service.py]: Should this be `error`?  `warn` means that you're able to proceed, despite the fact that something unexpected happened.

File: brkt_cli/gce/gce_service.py [hunk 3]
```diff
@@ -219,52 +219,32 @@ def __init__(self, project, session_id, logger):
...
     def validate_file_name(self, file):
-        m = re.match(r'[*?#]', file)
+        m = re.search(r'[*?#]', file)
         if m:
             raise ValidationError("File name can't contain '*','?'' or '#'")
-        if "[" or "]" in file:
-            raise ValidationError("File name cant contain '['' or ']'")
-        return 
+        if "]" in file or "[" in file:
+            raise ValidationError("File name cant contain '[' or ']'")
```
Comment (bburtin) [inline brkt_cli/gce/gce_service.py]: Typo.  Also, what about putting `[` and `]` into the previous check?  Seems like there's no reason to special-case them.

Comment (markvoll) [inline brkt_cli/gce/gce_service.py]: Consolidated into one check

File: brkt_cli/gce/gce_service.py (hunk 2 above)
Comment (markvoll) [inline brkt_cli/gce/gce_service.py]: Yes, switched to error

File: brkt_cli/gce/gce_service.py (hunk 1 above)
Comment (markvoll) [inline brkt_cli/gce/gce_service.py]: GCE does give an error code (409) but it gives the same error code if you own the bucket or you don't, only the messages are different.

**Pull request merged**